katcp/core.py

```diff
@@ -299,7 +299,23 @@ def format_argument(self, arg):
         elif isinstance(arg, bool):
             return str(int(arg))
         else:
-            return str(arg)
+            try:
+                return str(arg)
+            except Exception:
+                # unicode characters will break the str cast, so
+                # try to encode to ascii and replace the offending characters
+                # with a '?' character
+                logger.exception("Error casting message argument to str!")
+                try:
+                    if type(arg) != unicode:
+                        arg = arg.decode('utf-8')
+                    return arg.encode('ascii', 'replace')
+                except Exception:
+                    # If all else fails, ignore the string completely
+                    logger.exception(
+                        "Error decoding message argument from utf-8, "
```

Comment:
Not necessarily true. The failure could be in `arg.encode`, if `arg` isn't unicode.

Reply:
You're right, I changed it to `Error formatting argument as str, ignoring message and returning ''` which reflects the function name a bit better.